### DISCUSSION TITLE
Prompt with {[CLIPBOARD}} not working in Firefox

### DISCUSSION THREAD

**Discussion by btzs** [open-webui/open-webui#1659] [General]
# Bug Report

## Description

**Bug Summary:**
I set up a prompt including the ``{{CLIPBOARD}}`` tag to automatically include the clipboard. It works on Edge but not on Firefox. On Edge a dialogue opens to to ask for clipboard permission. On Firefox no such dialogue appears.

**Steps to Reproduce:**
1. Create normal prompt without ``{{CLIPBOARD}}`` and use it --> no problem
2. add ``{{CLIPBOARD}}`` to prompt --> prompt not working anymore. 
3. error in console

**Expected Behavior:**
- Prompt should be inserted into prompt field with {{CLIPBOARD}} getting replaced by the actual content of the clipboard. 
or:
- Warning that function is not supported on [Firefox](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/readText#browser_compatibility)

**Actual Behavior:**
Nothing happens.

## Environment

- **Open WebUI Version:** v0.1.119

- **Ollama (if applicable):** -

- **Operating System:** Windows 10
- **Browser (if applicable):** 
- not working on Firefox: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
- working on Edge: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0

## Reproduction Details
**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
```
Uncaught (in promise) TypeError: navigator.clipboard.readText is not a function
    Immutable 50
    <anonymous> https://xxxxx/:70
    promise callback* https://xxxxx/:69

```

**Docker Container Logs:**
-

## Installation Method

Docker

**Comment by tjbck** [comment]
Unable to reproduce the issue with FF 125.0.1, If anyone could reproduce the issue, PR welcome!

**Comment by jjczopek** [comment]
@btzs @tjbck IMO the issue is not with OpenWeb UI. When hosting using using docker, by default there is no SSL and according to this note on [developer.mozilla.com](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/readText), `navigator.clipboard.readText()` is available only in secure context (with SSL).

Same _issue_ happens on other browsers, like Chrome and Edge.

So it should be solved with adding SSL to OpenWebUI with e.g. reverse proxy.

**Reply by btzs** [reply]
I can confirm, the error disappeared once accessing via https.

**Reply by LumenYoung** [reply]
Hi,

I still face the same issue with SSL access to the openwebui instance and have navigator.clipboard.readText enabled in the about:config.

I wonder if there is any other possible cause to this problem?
